lib/services/spotify/save-user-info.ts

```diff
@@ -20,36 +20,36 @@ export const saveUserInfo = async ({
 }) => {
   const providerToken = session.provider_token;
   const userMetadata: UserMetadata = user.user_metadata;
+
   if (
-    providerToken &&
-    userMetadata.avatar_url &&
-    userMetadata.full_name &&
-    userMetadata.provider_id
+    !providerToken ||
+    !userMetadata.avatar_url ||
+    !userMetadata.full_name ||
+    !userMetadata.provider_id
   ) {
-    const cookieStore = cookies();
-    const supabase = createSupabaseServerClient(cookieStore);
-    const { error } = await supabase
-      .from('account')
-      .upsert(
-        {
-          avatar_url: userMetadata.avatar_url,
-          display_name: userMetadata.full_name,
-          provider_refresh_token: session.provider_refresh_token,
-          provider_token: providerToken,
-          spotify_id: userMetadata.provider_id,
-          user_id: user.id,
-        },
-        { onConflict: 'spotify_id' },
-      )
-      .select();
-
-    if (error) {
-      return NextResponse.json(
-        { message: 'Spotify into could not save into account table' },
-        {
-          status: 500,
-        },
-      );
-    }
+    throw new BadRequestError({
+      message: 'Missing user info',
+    });
+  }
+
+  const supabase = createSupabaseServerClient(cookies());
+
+  const { error } = await supabase
+    .from('account')
+    .upsert(
+      {
+        avatar_url: userMetadata.avatar_url,
+        display_name: userMetadata.full_name,
+        provider_refresh_token: session.provider_refresh_token,
+        provider_token: providerToken,
+        spotify_id: userMetadata.provider_id,
+        user_id: user.id,
+      },
+      { onConflict: 'spotify_id' },
+    )
+    .select();
+
+  if (error) {
+    throw new InternalServerError({ error });
```

Comment:
and this can be `DatabaseError` or we could just throw this and check `instanceof PostgreError` ? `new InternalServerError({ error: postgreError }).toNextResponse()` in the route handlers